### PR TITLE
Update Xcodeproj and cocoapods-deintegrate gems

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Update Xcodeproj and cocoapods-deintegrate gems  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
The issue mostly with `object_version` is that it is being treated (and should be) as an `Integer` across CocoaPods and Xcodeproj  but for some reason the `@object_version` ivar is stored and returned as a `String` from https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/project.rb#L82 and https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/project.rb#L124-L126.

This is public API so changing it now will be a breaking change................either way CocoaPods now correctly uses `Integer` across its classes but the documentation was not reflecting this properly.

File: spec/unit/installer/xcode/multi_pods_project_generator_spec.rb [hunk 1]
```diff
@@ -563,11 +563,11 @@ class Xcode
             target.stubs(:user_targets).returns([user_target])
 
             @generator = MultiPodsProjectGenerator.new(config.sandbox, [target], [],
-                                                       @analysis_result.all_user_build_configurations, @installation_options, config, '1')
+                                                       @analysis_result.all_user_build_configurations, @installation_options, config, 51)
             pod_generator_result = @generator.generate!
-            pod_generator_result.project.object_version.should == '1'
+            pod_generator_result.project.object_version.should == '51'
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/multi_pods_project_generator_spec.rb]: this is the annoying part from Xcodeproj that returns this value back as a `String`.

File: spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb [hunk 2]
```diff
@@ -85,7 +85,7 @@ class PodTargetIntegrator
               end
 
               it 'integrates test native targets with frameworks and resource script phase input and output file lists' do
-                @project.stubs(:object_version).returns(50)
+                @project.stubs(:object_version).returns('50')
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/pods_project_generator/pod_target_integrator_spec.rb]: this is more correct knowing the Xcodeproj returns a `String`

**Pull request merged**